### ISSUE TITLE
Can't open app from Heroku

### ISSUE THREAD

**Issue by 1rufaz** [Steem-FOSSbot/steem-fossbot-voter#93]
I just installed this again on heroku and when you go to open app all you get is this message:

version undefined
Session is valid, dashboard active
botty on Steemitbotty detailed data

**Comment by thrize** [comment]
> botty on Steemitbotty detailed data

I don't know what this means.

Sometimes when you open it for the first time the Heroku server takes a long time to boot up, because on free accounts they put everything to sleep when not being used and it takes time to wake up.

So have you tried just refreshing the web app using browser reload?

**Comment by 1rufaz** [comment]
Yes I have tried refreshing and installing.

When you click on open app from heroku, it's asks you to put in your api key, then I get this screen:

![2017-08-17](https://user-images.githubusercontent.com/29882131/29399936-1d4b88de-8323-11e7-9b4c-f5da1a917fd7.png)

**Comment by thrize** [comment]
Oh right, I get you. Hmm this is mysterious. I'll try a new install and see if I can replicate it.

**Comment by thrize** [comment]
No, it's is not replicating, no idea why it is happening like that for you. Make sure of the following things and try an new installation:

1. You are using Heroku install button the master branch of the main Voter project. Here it is in case [![Deploy](https://www.herokucdn.com/deploy/button.png)](https://heroku.com/deploy?template=https://github.com/Steem-FOSSbot/steem-fossbot-voter)
2. You enter the correct username, private posting key and a newly created bot API key

You should also re-read the [installation guide](https://github.com/Steem-FOSSbot/steem-fossbot-voter/blob/master/docs/installation.md) to make sure everything is followed.

**Comment by thrize** [comment]
@1rufaz don't forget to let me know if this works. It's a serious bug

**Comment by 1rufaz** [comment]
No worries, I have a feeling it's my heroku account, as I tried to install a previous version, that I know was working as i'd used it previously on the same heroku account and it did the same thing. I am just waiting for a response from heroku.

**Comment by 1rufaz** [comment]
I have just done a fresh install on another heroku account and I still get the same error.

**Comment by thrize** [comment]
What does it say in the logs?

**Comment by 1rufaz** [comment]
[botty1-logs-1503067639397.txt](https://github.com/Steem-FOSSbot/steem-fossbot-voter/files/1234747/botty1-logs-1503067639397.txt)

**Comment by thrize** [comment]
That's not really enough to go on I'm afraid. However I was able to see the URL of your Heroku app and try it.

It's very weird but it looks like it's failing in every way, I do not see why. For example Heroku is serving it as http, not https, so it comes up as insecure. Additionally it is not able to set or read the cookies, not able to get the version information from the package.json file, etc.

I don't know why it's happening and haven't heard of it from any other user. I presume there is something wrong with your account. Did you set yourself as a Node.js developer on that account? I think you need to contact Heroku support.

If you want I can continue to help if you post a much much larger logs file. Type in this into your command line to get more information: ```heroku logs --app botty1 --num 2000```

**Comment by 1rufaz** [comment]
Weird huh, I don't think it is heroku as I had the older version installed before and it was working fine, also I tried installing this version on another heroku account where I am running an older version, and I still had the same problem.

Thank you anyway.

**Comment by thrize** [comment]
Can you get in touch with Heroku support anyway?

**Comment by thrize** [comment]
I was able to replicate this randomly under the conditions I mentioned before.

> Sometimes when you open it for the first time the Heroku server takes a long time to boot up, because on free accounts they put everything to sleep when not being used and it takes time to wake up.

It worked whenever I refreshed the page. I really think this is to do with your account, that the virtual machine is not booting up quickly and perhaps sleeping immediately after fetch.

**Comment by 1rufaz** [comment]
Interesting! I have tried refreshing and refreshing and it doesn't work on this account. I did email heroku support this morning and I am still waiting on a reply. I will keep you updated.

**Comment by 1rufaz** [comment]
I'm still waiting for a reply from Heroku

**Comment by 1rufaz** [comment]
Funny, I just put my steemit account details into a version I have working for another account, and I got the same problem???!!!???

**Comment by 1rufaz** [comment]
So it's not a Heroku problem as I put different account details into the heroku account that had the problem and it worked fine.

**Comment by thrize** [comment]
I wonder if it could be the same but as #102, does the account you're using have a period in it or any special character?

**Comment by 1rufaz** [comment]
Interesting that someone else is having the same problem but my steemit name has no special characters.

**Comment by scottiesan** [comment]
i am getting the same issue.  Is happening on docker and heroku.

**Comment by braydengerrard** [comment]
I just had the same issue.  After several refreshes, I am now getting the Application Error page instead.  Checking logs to see what they say.

**Comment by bobinson** [comment]
+1

**Comment by thrize** [comment]
No idea why it's happening, give me more logs if you can, @bobinson that isn't enough information. I'll look at this after finishing the MongoDB migration

**Comment by bobinson** [comment]
[indiaunited-logs-1515329711913.txt](https://github.com/Steem-FOSSbot/steem-fossbot-voter/files/1609764/indiaunited-logs-1515329711913.txt)
@thrize  - more logs uploaded above. (I have removed an email address from it.)

1. I also have a Docker deployment and if you can help me with where to look for the logs, I can share that too.

**Comment by thrize** [comment]
Thanks. I'm not in a position to look at this right now but add any details and I will review as soon as I can. I'm getting maintenance work back up running for this app.

**Comment by bobinson** [comment]
@thrize  - does it has anything to do with Steem Inc changing their API node from https://steemd.steemit.com

to

https://api.steemit.com

?

**Comment by braydengerrard** [comment]
My app has crashed.  This is the error I am getting in logs:

2018-01-07T22:53:17+00:00 app[heroku-redis]: source=REDIS_URL sample#used_memory_over_limit=2367048.0bytes message=Database memory close to the limit. Please upgrade your database plan or change your maxmemory policy.

I don't quite understand this.  What memory do I have to upgrade?

**Comment by thrize** [comment]
@braydengerrard that's off topic, it's a different bug and will be fixed anyway by the Mongo DB migration tracked in #64

**Comment by thrize** [comment]
Looks like it's to do with this issue on steem-js https://github.com/steemit/steem-js/issues/312

Trying to get a fix

**Comment by thrize** [comment]
It looks like the master branch is not actually functioning, I've tried all kinds of different ways to use when built from GitHub directly instead of NPM but the module is not registered correctly. If the steem-js guys do not respond soon I'll fork steem-js and try to fix it manually until they get on it but I really hope to not have to do that!

TLDR; still waiting

**Comment by bobinson** [comment]
@thrize - I think they are going to be busy with the Meltdown and SPECTRE patches and in the meantime its a good idea to find a workaround including a temporary fork? - which they can backmerge.

**Comment by thrize** [comment]
I'll hold on for a few hours as I'm not available to work on it until later anyway but you might be right.

Anyone who wants can get started on this if you have the skill and time.

**Comment by bobinson** [comment]
I would love to take up but I am not experienced with JS

**Comment by Milesolson** [comment]
I am having this same problem

**Comment by thrize** [comment]
Everyone, this is temporarily working again, see my comment on #102

**Comment by thrize** [comment]
Sorry guys, I didn't realize there were other websocket services still available. I'll pointed ours to gandalf's wss and made a patch release 0.2.10

You can upgrade using the usual way. Closing as this is fixed. When Steem-js is updated I'll move over to that but there should be no interruption in service, fingers crossed.

**Comment by bobinson** [comment]
Just thought, will it be possible to check the api.steemit.com is responding and if not use another one like gtg's or buildteam's ?

**Comment by thrize** [comment]
Yes I'm sure that's possible. Why don't you log it as a request? Strictly it's out of scope of this ticket

**Comment by thrize** [comment]
Hoping to resolve this this week

**Comment by thrize** [comment]
Mainly resolved, as of #64 

Note that you still might get this if the Heroku dyno was idle and had to start up. #21 (improving server start up time) should resolve this completely, so this issue still blocked.

**Comment by thrize** [comment]
#21 is done and in review, review this an mark as done when confirmed start time is quick enough to allow server to deliver full dashboard web page from idle Heroku dyno

**Comment by thrize** [comment]
Confirmed fixed

**Comment by 1rufaz** [comment]
This issue is still not resolved for me, is there anyway I can message you?

**Comment by thrize** [comment]
Good @1rufaz !
